pkg/core/interop/binary/itoa.go

```diff
@@ -0,0 +1,93 @@
+package binary
+
+import (
+	"encoding/hex"
+	"errors"
+	"math/big"
+	"strings"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/interop"
+	"github.com/nspcc-dev/neo-go/pkg/encoding/bigint"
+)
+
+var (
+	// ErrInvalidBase is returned when base is invalid.
+	ErrInvalidBase = errors.New("invalid base")
+	// ErrInvalidFormat is returned when string is not a number.
+	ErrInvalidFormat = errors.New("invalid format")
+)
+
+// Itoa converts number to string.
+func Itoa(ic *interop.Context) error {
+	num := ic.VM.Estack().Pop().BigInt()
+	base := ic.VM.Estack().Pop().BigInt()
+	if !base.IsInt64() {
+		return ErrInvalidBase
+	}
+	var s string
+	switch b := base.Int64(); b {
+	case 10:
+		s = num.Text(10)
+	case 16:
+		if num.Sign() == 0 {
+			s = "0"
+			break
+		}
+		bs := bigint.ToBytes(num)
+		reverse(bs)
+		s = hex.EncodeToString(bs)
+		switch {
+		case bs[0]&0xF0 == 0 && bs[0]&0x08 == 0:
+			s = s[1:]
+		case bs[0]&0xF0 == 0xF0 && bs[0]&0x08 != 0:
+			s = s[1:]
+		}
+	default:
+		return ErrInvalidBase
+	}
+	ic.VM.Estack().PushVal(strings.ToUpper(s))
```

Comment:
ToUpper is only relevant for `base == 16`, there is no need to invoke it for `base == 10`.

Reply:
Fixed.